### ISSUE TITLE
react/button-has-type

### ISSUE THREAD

**Issue by andrevenancio** [yannickcr/eslint-plugin-react#2020]
this rule PASSES
`<button type="button">Hi</button>`

this rule FAILS
`<button>Hi</button>`

But this rule should pass. Right?
```
const customProps = {
    type: 'button',
};
<button {...customProps}>Hi</button>
```

**Comment by vladshcherbin** [comment]
@ljharb 

This one also doesn't work:

```js
const buttonType = 'button'

return (
  <button type={buttonType}>hey</button>
)
```

The error in this case is: *\`buttonType\` is an invalid value for button type attribute*

So it's not possible to pass type as a prop and don't have this rule failing.
